### PR TITLE
Forms: update Time input class, update DatabaseFormFactory

### PR DESCRIPTION
**Pull request by SKuipers**
A couple tweaks to the form classes. Primarily adds minimum and maximum methods to the Time input class, used to limit the available times client-side.

### CODE REVIEW

File: src/Forms/Input/Time.php [hunk 1]
```diff
@@ -42,14 +45,38 @@ public function setFormat($format)
...
 
+    /**
+     * Define a minimum for this time value.
+     * @param   string  $value
+     * @return  self
+     */
+    public function minimum($value)
+    {
+        $this->min = $value;
+        return $this;
```
Comment (rossdotparker) [inline src/Forms/Input/Time.php]: Why is this not:

`return $this->min;`

File: src/Forms/Input/Time.php [hunk 2]
```diff
@@ -42,14 +45,38 @@ public function setFormat($format)
...
+
+    /**
+     * Define a maximum for this time value.
+     * @param   string  $value
+     * @return  self
+     */
+    public function maximum($value)
+    {
+        $this->max = $value;
+        return $this;
```
Comment (rossdotparker) [inline src/Forms/Input/Time.php]: Likewise, why is this not:

`return $this->max;`

**Comment by rossdotparker** [comment]
I dropped a couple questions, but happy enough to merge in the meantime. I'd definitely be interested in learning the answer. Is it to do with the way the method calls can be chained together?

**Comment by SKuipers** [comment]
Yep! They're a chained method call, so it sets a value and returns itself, so the chain can continue. I've tried to make all the setters in the Form and Table APIs chainable, to give us the ability to compose objects in a readable fashion.

**Comment by jimbo8098** [comment]
That's a good idea, I like it 👍

**Pull request merged**